### ISSUE TITLE
Step-by-step navigation does not convey all of the information for screen reader users when viewing them as forms

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-design-system#3021]
This issue is from May 2023 external accessibility audit report.

**Usability issue**

**Issue ID:**
DAC_Usability_Step_By_Step_01

**URLs:**
[https://www.gov.uk/visit-uk-business-trip](https://www.gov.uk/visit-uk-business-trip)

## Screen shots
<img width="773" alt="step-by-step navigation with 'or' steps" src="https://github.com/alphagov/govuk-frontend/assets/56260216/dae3f223-4741-4984-a24d-44d786e4e27a">

The step by step navigation page consists of headings containing buttons as the components used to expand the content. However, the step number content is not included within the button elements. This means that the step information is only conveyed when viewing them as headings and not as forms.

## Current code ref(s)
`#apply-for-a-family-permit`

## Screen reader comments
> “I found that there were multiple steps in the process with both the step number and accordion buttons marked-up as headings. 

>JAWS and NVDA: 
With the desktop screen reader software, I found that when using headings shortcut commands such as the ‘h’ key both the step and the accordion appeared within the same heading announcing as a single item. In context, however, these items appeared to be separated on different lines, although the content was clear when using the headings. When using form element shortcuts, however, the step did not announce within the label of the accordion announcing only the category. While situated out of context I was not able to easily understand the order of the steps and could only view the category. I would have found it easier if the content located within the heading text could also be found within the label of the accordion as this would enable me to immediately understand what is occurring both in and out of context. 

> VoiceOver: 
When I moved in context, I found that the ‘Step’ heading appeared over two separate lines. I needed to swipe three times past each category to move to the next category with two swipes needed to move past the heading and then the third to move past the category. This content made the structure more difficult to understand although I was not prevented from understanding the content as a highly experienced screen reader user. When using the forms rotor like desktop screen reader software I was not able to locate the step number with only the category announcing therefore making the order more problematic to understand. 

> TalkBack: With TalkBack I found that the step and category announced over multiple lines. When using the ‘Controls’ menu I was only able to locate the name of the category meaning the order was not clear. I also found that the headings relating to the step and accordion were not detected when swiping in context. Instead, I needed to use the headings rotor to locate the headings meaning that in a non-testing environment some screen reader users may not locate the headings and could become confused.”

## Solution
Consider incorporating all of the content and information into the <button> element, allowing for users of screen reading assistive technologies to have that information announced and conveyed with the label of the form, and not just the heading.
